Ability to generate passwords using diceware

raphael247 opened the pull request:
To generate dice-passwords:

- set the option RANDOM_PWD=dice in school.conf
- PWD_LENGTH will be the number of words separated by dots.
- diceware must be installed by 'pip3 install diceware'
- Samba must be configured to permit non-komplex passwords via 'samba-tool domain passwordsettings set --complexity=off'

Commit:
Ability to generate passwords using diceware

To generate dice-passwords:
- set the option RANDOM_PWD=dice in school.conf
- PWD_LENGTH will be the number of words separated by dots.
- diceware must be installed by 'pip3 install diceware'
- Samba must be configured to permit non-komplex passwords via
  'samba-tool domain passwordsettings set --complexity=off'

toheine commented:
What is the current status of this PR? This feature makes absolutely sense. Please merge or get in touch with @raphael247 if this should be implemented differently.

anschuetz commented:
Hi @raphael247,
hi @toheine,

We apologize for the delay in reviewing your pull request. Unfortunately, the developer responsible for this has been unavailable for an extended period. However, we've addressed the situation by creating a fork to continue the process, and we'll be moving forward with the necessary steps to handle your contribution.

Thank you for your patience and understanding.

Best regards,
Jesko

Commit:
Config options for DICE passwords and improved error handling

Introduced DICE config options in school.conf:
[userfile.students|teacher|extrastudents.csv]
    DICE_LANG = de_8k
    DICE_SEPARATOR = .
    DICE_WORDS = 2
    DICE_CAPS = TRUE

Error checks:
- diceware package not installed
- samba password complexity not disabled
- diced password too short (samba minimum password length)
